src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -354,16 +356,22 @@ private static bool CompileProject(ProjectContext context, CompilerCommandApp ar
             }
             else if (!string.IsNullOrEmpty(context.ProjectFile.TestRunner))
```

Comment:
discussed offline. Let's make this a first-class compiler option that's set true by default for test projects and executables.